cmd/btfs/daemon.go

```diff
@@ -299,6 +297,10 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 	pubsub, _ := req.Options[enablePubSubKwd].(bool)
 	mplex, _ := req.Options[enableMultiplexKwd].(bool)
 
+	// Btfs auto update.
+	url := fmt.Sprint("localhost:", strings.Split(cfg.Addresses.API[0], "/")[4])
```

Comment:
Since a multiaddr looks like this `/ip4/0.0.0.0/tcp/5001`

Can you use 

`url := strings.Split(cfg.Addresses.API[0], "/")[2] + ":" + strings.Split(cfg.Addresses.API[0], "/")[4]`? So it uses the setting's host as well (sometimes complex network setups don't always work with localhost)

Reply:
@SiyingPeng

Reply:
done

Reply:
@taiyangc